blocks/library/html/index.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import TextareaAutosize from 'react-autosize-textarea';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { Component } from 'element';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlockType, query } from '../../api';
+import BlockControls from '../../block-controls';
+
+const { children } = query;
+
+registerBlockType( 'core/html', {
+	title: __( 'HTML' ),
+
+	icon: 'editor-code',
+
+	category: 'formatting',
+
+	attributes: {
+		content: children(),
+	},
+
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+			this.preview = this.preview.bind( this );
+			this.edit = this.edit.bind( this );
+			this.state = {
+				preview: false,
+			};
+		}
+
+		preview() {
+			this.setState( { preview: true } );
+		}
+
+		edit() {
+			this.setState( { preview: false } );
+		}
+
+		render() {
+			const { preview } = this.state;
+			const { attributes, setAttributes, focus } = this.props;
+
+			return (
+				<div>
+					{ focus &&
+						<BlockControls key="controls">
+							<ul className="components-toolbar">
```

Comment:
Can't we use the regular `Toolbar` component for this? I guess we just need to make the icon optional and drop the fixed width?